distributed/diagnostics/plugin.py

```diff
@@ -301,18 +310,9 @@ async def setup(self, worker):
                 await self._set_restarted(worker)
                 worker.loop.add_callback(worker.close_gracefully, restart=True)
 
-    def _install(self):
-        proc = subprocess.Popen(
-            [sys.executable, "-m", "pip", "install"] + self.pip_options + self.packages,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        _, stderr = proc.communicate()
-        returncode = proc.wait()
-        if returncode != 0:
-            msg = f"Pip install failed with '{stderr.decode().strip()}'"
-            logger.error(msg)
-            raise RuntimeError(msg)
+    @abc.abstractmethod
+    def _install(self) -> None:
```

Comment:
As now somebody could legitimately tap into this to make e.g. a Poetry installer, could you make all abstractmethods public?

Reply:
This should only be called internally during setup, i.e., by the subclass itself, not the user. This is why I made it private since we lack a nuanced way of access control. I'm open to making it public though.